src/components/landing/ForgotPassword.jsx

```diff
@@ -1,32 +1,53 @@
 import React, { useState } from 'react';
+import { notification } from 'antd';
 import PropTypes from 'prop-types';
 import LandingCard from './LandingCard.jsx';
 import TextInput from '../TextInput.jsx';
 import Button from '../Button.jsx';
 import '../../scss/landing.scss';
+import API from '../../api/API';
 
 /// callBack should be passed from landing page. Used to re-render the landing page
 /// with the default landing page card, which should be the login card.
 
 function ForgotPassword({ switchCard }) {
-  let input;
+  // Leaving input as stateless since it should probably reset if user goes back
+  // to login.
+  let input = '';
   const [err, setError] = useState(null);
 
-  function sendResetEmail(event) {
+  function isTrimmedEmpty(str) {
+    if (str.trim() === '') {
+      return true;
+    }
+    return false;
+  }
+
+  async function onSubmit(event) {
     event.preventDefault();
-    // TODO: Make reset password request
-    setError('Not Implemented');
+    try {
+      if (isTrimmedEmpty(input) === false) {
+        await API.passwordRecovery(input);
+        notification.success({
+          description: 'Please check your email for the password reset link.',
+        });
+        setError('');
+      } else {
+        throw (new Error('Email field cannot be empty.'));
+      }
```

Comment:
Just an opinion but I think it'd be better to just disable the submit button if the input is empty rather than showing an error for an empty input.

Reply:
Good point.